src/v/datalake/tests/record_multiplexer_test.cc

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "datalake/record_multiplexer.h"
+#include "datalake/tests/test_data_writer.h"
+#include "storage/tests/storage_test_fixture.h" // Using storage_fixture_test to easily create a test log
+#include "test_utils/fixture.h"
+
+#include <seastar/util/defer.hh>
+
+#include <boost/test/tools/old/interface.hpp>
+
+FIXTURE_TEST(record_multiplexer_test, storage_test_fixture) {
+    auto cfg = default_log_config(test_dir);
+    cfg.max_segment_size = config::mock_binding<size_t>(10 * 1024);
+    storage::log_manager mgr = make_log_manager(std::move(cfg));
+    info("Configuration: {}", mgr.config());
+    auto deferred = ss::defer([&mgr]() mutable { mgr.stop().get0(); });
+    auto ntp = model::ntp("default", "test", 0);
+    auto log
+      = mgr.manage(storage::ntp_config(ntp, mgr.config().base_dir)).get0();
+
+    // Append some linear kv ints.
+    int num_batches = 5;
+    append_random_batches<linear_int_kv_batch_generator>(log, num_batches);
+    log->flush().get0();
+
+    // Validate
+    auto batches = read_and_validate_all_batches(log);
```

Comment:
Can we write a test that doesn't use an entire fixture? For example you could use like an in-memory batch reader instead of a fully materialized log.